crates/compiler/test_gen/src/gen_list.rs

```diff
@@ -1328,9 +1328,30 @@ fn list_map_all_inline() {
     );
 }
 
+#[test]
+#[cfg(any(feature = "gen-llvm", feature = "gen-wasm", feature = "gen-dev"))]
+fn list_map_closure_int() {
+    assert_evals_to!(
+        indoc!(
+            r#"
+            int : I64
+            int = 123
+
+            single : List I64
+            single =
+                [0]
+
+            List.map single (\x -> x + int)
+            "#
+        ),
+        RocList::from_slice(&[123]),
+        RocList<i64>
+    );
+}
+
 #[test]
 #[cfg(any(feature = "gen-llvm", feature = "gen-wasm"))]
-fn list_map_closure() {
+fn list_map_closure_float() {
```

Comment:
this float test has some issue, the int test above succeeds. So I suspect there is some problem with how we handle float registers in the dev backend.

Reply:
I would guess it is in function argument setting/loading, but that is a very loose guess.